tests/filecheck/dialects/csl/ops.mlir

```diff
@@ -63,11 +63,35 @@ csl.func @initialize() {
 
     %col_1 = "csl.get_color"() <{id = 3 : i5}> : () -> !csl.color
 
+
+    %arr, %scalar = "test.op"() : () -> (memref<10xf32>, i32)
+
+    %scalar_ptr = "csl.addressof"(%scalar) : (i32) -> !csl.ptr<i32, #csl<ptr_kind single>, #csl<ptr_const const>>
+    %many_arr_ptr = "csl.addressof"(%arr) : (memref<10xf32>) -> !csl.ptr<f32, #csl<ptr_kind many>, #csl<ptr_const const>>
+    %single_arr_ptr = "csl.addressof"(%arr) : (memref<10xf32>) -> !csl.ptr<memref<10xf32>, #csl<ptr_kind single>, #csl<ptr_const const>>
+
+
+
   csl.return
 }
+
+%other_global_ptr = "test.op"() : () -> !csl.ptr<i16, #csl<ptr_kind single>, #csl<ptr_const var>>
+
+"csl.export"() <{var_name = @initialize, type = () -> ()}> : () -> ()
+"csl.export"() <{var_name = @global_ptr, type = !csl.ptr<f32, #csl<ptr_kind many>, #csl<ptr_const const>>}> : () -> ()
```

Comment:
Where is `@global_ptr`? I think we should only allow function exporting using symbols, or do we have an op to "bind" a value to a symbol?